core/data/src/main/java/com/teamwable/data/repositoryimpl/DefaultCommentRepository.kt

```diff
@@ -0,0 +1,63 @@
+package com.teamwable.data.repositoryimpl
+
+import androidx.paging.Pager
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
+import androidx.paging.map
+import com.teamwable.data.mapper.toData.toPostCommentDto
+import com.teamwable.data.mapper.toData.toPostGhostDto
+import com.teamwable.data.mapper.toModel.toComment
+import com.teamwable.data.repository.CommentRepository
+import com.teamwable.model.Comment
+import com.teamwable.model.Ghost
+import com.teamwable.network.datasource.CommentService
+import com.teamwable.network.util.GenericPagingSource
+import com.teamwable.network.util.handleThrowable
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import javax.inject.Inject
+
+class DefaultCommentRepository @Inject constructor(
+    private val apiService: CommentService,
+) : CommentRepository {
+    override fun getHomeDetailComments(feedId: Long): Flow<PagingData<Comment>> {
+        return Pager(PagingConfig(pageSize = 15, prefetchDistance = 1)) {
+            GenericPagingSource(
+                apiCall = { cursor -> apiService.getHomeDetailComments(feedId, cursor).data },
+                getNextCursor = { comments -> comments.lastOrNull()?.commentId },
+            )
+        }.flow.map { pagingData ->
+            pagingData.map { it.toComment() }
+        }
+    }
+
+    override fun getProfileComments(userId: Long): Flow<PagingData<Comment>> {
+        return Pager(PagingConfig(pageSize = 10, prefetchDistance = 1)) {
+            GenericPagingSource(
+                apiCall = { cursor -> apiService.getProfileComments(userId, cursor).data },
+                getNextCursor = { comments -> comments.lastOrNull()?.commentId },
+            )
+        }.flow.map { pagingData ->
+            pagingData.map { it.toComment() }
+        }
+    }
+
+    override suspend fun deleteComment(commentId: Long): Result<Boolean> = runCatching {
+        apiService.deleteComment(commentId).success
+    }.onFailure {
+        return it.handleThrowable()
+    }
```

Comment:
p4 : 
```suggestion
    override suspend fun deleteComment(commentId: Long): Result<Unit> = runCatching {
        apiService.deleteComment(commentId)
        Unit
    }.onFailure {
        return it.handleThrowable()
    }
```
단순하게 상위에서 서버 통신 성공/실패 여부만 필요 하다면 Boolean리턴할 필요는 없을 것 같아요!
성공 실패 여부는 success, failure 블록안에 담겨가요!

Reply:
!!! 이거 진짜 좋은거 같아요 요 생각을 못했었네.. 쇽샥
[24c8a14](https://github.com/Team-Wable/WABLE-ANDROID/pull/58/commits/24c8a148a5a892340576cc85c07cb4bc59bf124b)